src/main/java/com/zaxxer/nuprocess/osx/ProcessKqueue.java

```diff
@@ -192,59 +261,38 @@ else if (WIFSIGNALED(status)) {
             osxProcess.onExit(status);
          }
       }
-      else if (filter == Kevent.EVFILT_SIGNAL) {
-         checkStdinCloses();
-         checkWaitWrites();
-      }
-
-      kevent.clear();
-
-      return true;
    }
 
    // ************************************************************************
    //                             Private methods
    // ************************************************************************
 
-   private void queueEvent(int handle, int filter, int flags, int fflags, Pointer udata)
-   {
-      try {
-         Kevent kevent = keventPool.take();
-
-         Kevent.EV_SET(kevent, (long) handle, filter, flags, fflags, 0l, udata);
-         LibKevent.kevent(kqueue, kevent.getPointer(), 1, null, 0, null);
-
-         keventPool.put(kevent);
-      }
-      catch (InterruptedException e) {
-         return;
-      }
-   }
-
    private void checkStdinCloses()
    {
-      if (closeQueue.isEmpty()) {
-         return;
-      }
-
-      OsxProcess process;
-      while ((process = closeQueue.poll()) != null) {
-         process.stdinClose();
-      }
+     List<OsxProcess> processes = new ArrayList<OsxProcess>();
+     closeQueue.drainTo(processes);
```

Comment:
The trouble with ``drainTo()`` is that it is not atomic.  The JavaDoc states: ``the behavior of this operation is undefined if the specified collection is modified while the operation is in progress.``  In practical terms, it is possible for a process to be in both the drained list and yet still remain in the queue if an element is added during the ``drainTo()`` operation.

The efficiency gain here is likely not measurable in real world use, so I recommend leaving this method as is.